system/src/system_power_manager.cpp

```diff
@@ -961,7 +978,7 @@ int PowerManager::getConfig(hal_power_config* conf) {
 
 #if HAL_PLATFORM_POWER_MANAGEMENT_OPTIONAL
     if (!err) {
-      if (flags & HAL_POWER_PMIC_DETECTION) {
+      if (!(flags & HAL_POWER_PMIC_DETECTION)) {
```

Comment:
> @scott-brust This doesn't look right. `hal_power_load_config()` returns flags normally as 1 - enabled, 0 - disabled and preprocesses things internally for DCT weirdness.

I see what you mean. msom was not detecting pmic without this change, ill look closer at whats happening in `hal_power_load_config()` and in DCT

Reply:
Most likely somebody/some tool/application wrote into DCT directly without too much care for other flags. If you have a freshly manufactured device, might be worth dumping DCT.

Reply:
My understanding is for whatever reason that breaks the power config, we'll restore the default config. Since this block is bracheted by `HAL_PLATFORM_POWER_MANAGEMENT_OPTIONAL`, so we should restore the default flag, which should set `HAL_POWER_PMIC_DETECTION`.

Reply:
> My understanding is for whatever reason that breaks the power config, we'll restore the default config. Since this block is bracketed by `HAL_PLATFORM_POWER_MANAGEMENT_OPTIONAL`, so we should restore the default flag, which should set `HAL_POWER_PMIC_DETECTION`.

Right, this could be simplified to:
```
#if HAL_PLATFORM_POWER_MANAGEMENT_OPTIONAL
    conf->flags |= HAL_POWER_PMIC_DETECTION;
#else
    (void)flags;
#endif // HAL_PLATFORM_POWER_MANAGEMENT_OPTIONAL
```
The problem is when DCT is unset, the `hal_power_config* conf` stored there is all `0xFF`. `hal_power_load_config()` tries to ensure the `HAL_POWER_PMIC_DETECTION` flag is set, but its not working:
```
        // Invert first byte of the flags to keep compatibility for HAL_POWER_PMIC_DETECTION flag
        uint32_t inverted = (~conf->flags) & 0x000000ff;
        conf->flags &= 0xffffff00;
        conf->flags |= inverted;
```
If `conf->flags` is `0xFFFFFFFF`, then `inverted` is just `0x00000000`. This function is returning with `conf->flags` set to `0xffffff00`, ie the pmic detect bits are not set. This is why I had to set them in `system_power_manager`

I logged the result in `PowerManager::getConfig()` and confirmed this behavior:
```
[2023-07-31 11:09:29.268] 0000000633 [sys.power] WARN: err 0 conf->version ff conf->flags ffffff00
```

Reply:
I think this failsafe is expected DCT to be erased/zeroed, not un-set/0xFF

Reply:
To wrap this up: the lower byte of flags in DCT needs to maintain backwards compatibility with previous DVOS versions / DCT structures. In these versions, clearing the bit (ie setting to 0) would *enable* the feature. This is why we invert the bits in `hal_power_load_config()`. 
When system power manager looks at these flags from DCT, it should only enable PMIC detection if this is explicitly enabled in DCT. 
EVT devices, which do not have anything in DCT, will return 0xFF in that byte, which indicates the PMIC detection is *DISABLED*. This is the correct default state for SoMs.
If a user wants to enable PMIC detection, it must be explicitly enabled. For MSoM EVT, we likely do want to enable this feature to allow the SOM + EVB board to run with only USB power
The easiest way to enable this is to run the `tinker` application, as it will enable the PMIC detection for SOM platforms by default.